### PR TITLE
SPORTSHUB-172 Mobile Search Fix

### PR DESCRIPTION
**Pull request by Edwin-Ni**
## Why was your change needed?

**!! There is a currently a bug with location in search, so I have made changes to disable the field for now !!**
Mobile search was broken

## What was Changed?

Added search functionality
Fixed recent searches

## Any extra notes for reviewers to know?

Recent search works by adding an entry only when the search button is clicked or the 'enter' key is pressed.
So a recent search is not added when the recent search entry is clicked on or if visiting from an URL.

### CODE REVIEW

File: frontend/components/mobile/MobileSearchInput.tsx [hunk 1]
```diff
@@ -84,34 +117,38 @@ export default function MobileSearchInput(props: MobileSearchInputProps) {
...
+            value={location}
             onChange={(event) => {
-              setSearchLocation(event.target.value);
+              setLocation(event.target.value);
             }}
+            onKeyDown={handleKeyPress}
           />
         </div>
-        <div className="w-full mt-14 ml-1">
+        {/* <div className="w-full mt-14 ml-1">
```
Comment (ReggieStar11) [inline frontend/components/mobile/MobileSearchInput.tsx:128]: remove this

Comment (Edwin-Ni) [reply frontend/components/mobile/MobileSearchInput.tsx:128]: We want to implement tags in the future still, so I think we can leave it there

**Review by ReggieStar11 — APPROVED** [review]

**Review by ReggieStar11 — CHANGES_REQUESTED** [review]
searches seem one search behind

Edwin - Searches are behind when location param is used, not an issue from my changes but already existing in master

File: frontend/components/mobile/MobileSearchInput.tsx [hunk 2]
```diff
@@ -55,63 +101,66 @@ export default function MobileSearchInput(props: MobileSearchInputProps) {
...
+        </div> */}
         <div className="w-full mt-10 ml-1">
           <h3 className="font-semibold text-lg">Recent Searches</h3>
-          <div className="h-36 overflow-y-scroll">
+          <div className="h-44 overflow-y-scroll">
             {recentSearches.length === 0 ? (
-              <p className="text-sm font-light">No recent searches... go ahead ;)</p>
+              <p className="text-sm font-light">None</p>
             ) : (
+              // recentSearches.map((search, i) => {
```
Comment (ReggieStar11) [inline frontend/components/mobile/MobileSearchInput.tsx:140]: Delete if unused

**Review by ReggieStar11 — APPROVED** [review]

**Pull request merged**